lib/index.js

```diff
@@ -875,6 +889,44 @@ exports.filter = (() => {
 	}
 })()
 
+/**
+ * @summary Compiles schema checking the cache
+ * @function
+ * @private
+ *
+ * @param {Object} ajvInstance - an instance of ajv
+ * @param {Object} schema - the schema to compile
+ * @returns {Function} result of ajvInstance.compile(schema)
+ *
+ * @example
+ * compileSchemaCached(ajv, {
+ *	type: 'object',
+ *	properties: {
+ *		foo: {
+ *			type: 'number'
+ *		}
+ *	},
+ *	required: [ 'foo' ]
+ *	})
+ *
+ */
+const compileSchemaCached = (ajvInstance, schema) => {
```

Comment:
This will return a different function depending on which ajvInstance you pass to it

Reply:
@LucianBuzzo you are right, this was potentially very wrong :P I refactored to make it sane